src/current/v23.1/goldengate.md

```diff
@@ -0,0 +1,530 @@
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    SET CLUSTER SETTING sql.defaults.datestyle.enabled = true;
+    SET CLUSTER SETTING sql.defaults.intervalstyle.enabled = true;
+    ~~~
+
+    For versions of CockroachDB v22.2 and later, these settings are already enabled by default.
+
+- Ensure [libpg](https://www.postgresql.org/download/linux/redhat/) is available on the Oracle GoldenGate host.
+
+- Ensure you have a secure, publicly available CockroachDB cluster running the latest **{{ page.version.version }}** [production release](https://www.cockroachlabs.com/docs/releases/{{ page.version.version }}), and have created a [SQL user]({% link {{ page.version.version }}/security-reference/authorization.md %}#sql-users).
+
+## Configure Oracle GoldenGate for PostgreSQL
+
+This section describes how to configure Oracle GoldenGate for PostgreSQL to work with CockroachDB. Oracle GoldenGate for PostgreSQL is the process that pulls data from [trail files](https://docs.oracle.com/goldengate/c1230/gg-winux/GGCON/processes-and-terminology.html) over to CockroachDB. Note that this is a separate installation from Oracle GoldenGate for Oracle, which will be used to pull source data and route it to proper [trail files](https://docs.oracle.com/goldengate/c1230/gg-winux/GGCON/processes-and-terminology.html). For more information, refer to the [Oracle GoldenGate for PostgreSQL documentation](https://docs.oracle.com/en/middleware/goldengate/core/19.1/gghdb/preparing-database-oracle-goldengate-postgresql.html). The following example uses a [CockroachDB {{ site.data.products.serverless }} cluster](https://cockroachlabs.com/docs/cockroachcloud/create-a-serverless-cluster).
+
+1. Edit the following parameters in the `ODBC.ini` file.
+
+    Set up the ODBC data sources and configuration:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~
+    # No changes should be needed for CRDBLOCAL
+    [ODBC Data Sources]
+    PG_src=DataDirect 7.1 PostgreSQL Wire Protocol
+    PG_tgt=DataDirect 7.1 PostgreSQL Wire Protocol
+    CRDB=DataDirect 7.1 PostgreSQL Wire Protocol
+
+    [ODBC]
+    IANAAppCodePage=4
+    # The following path is your Oracle GoldenGate for PostgreSQL installation directory
+    InstallDir=/u01/ggs-pg
+    ~~~
+    
+1. Ensure that all the PostgreSQL libraries are installed and referenced in `LD_LIBRARY_PATH`. 
+
+    The path should at least include `/usr/pgsql-13/lib` and `/u01/ggs-pg/lib`:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    # The path is a concatenation of your PostgreSQL libraries and GoldenGate installation directory
+    export LD_LIBRARY_PATH=/usr/pgsql-13/lib:/u01/ggs-pg/lib
+    ~~~
+
+1. Set up the `ODBC.ini` file for the Oracle GoldenGate host:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    # This is needed so that OGG knows where to look for connection details for the database
+    export ODBCINI=/etc/odbc.ini
+
+    vi /etc/odbc.ini
```

Comment:
Is the last line a stray?

Reply:
@ryanluu12345

Reply:
I think you intend for them to open the odbc.ini in their editor, but if they copy-paste this code block, they will unexpectedly be in `vi`, which is like Hotel California. I suggest splitting the editing into a separate step and not specifying the editor.